src/accountsdb/two/Two.zig

```diff
@@ -0,0 +1,175 @@
+//! Thin wrapper around Rooted and Unrooted that ties everything together to be used by the account store.
+
+const std = @import("std");
+const builtin = @import("builtin");
+const sig = @import("../../sig.zig");
+pub const Rooted = @import("Rooted.zig");
+pub const Unrooted = @import("Unrooted.zig");
+
+const Pubkey = sig.core.Pubkey;
+const Slot = sig.core.Slot;
+const Ancestors = sig.core.Ancestors;
+const Account = sig.core.Account;
+const AccountSharedData = sig.runtime.AccountSharedData;
+
+const Db = @This();
+
+allocator: std.mem.Allocator,
+rooted: Rooted,
+unrooted: Unrooted,
+
+pub fn init(allocator: std.mem.Allocator, rooted: Rooted) !Db {
+    var unrooted: Unrooted = try .init(allocator);
+    errdefer unrooted.deinit(allocator);
+
+    return .{
+        .allocator = allocator,
+        .rooted = rooted,
+        .unrooted = unrooted,
+    };
+}
+
+/// Initializes a temporary empty rooted storage. Call tmp.cleanup() when done with it.
+pub fn initTest(allocator: std.mem.Allocator) !struct { Db, std.testing.TmpDir } {
+    if (!builtin.is_test) @compileError("only used in tests");
+
+    const tmp = std.testing.tmpDir(.{});
+    var tmp_dir_buffer: [std.fs.max_path_bytes]u8 = undefined;
+    const tmp_path = try tmp.dir.realpath(".", &tmp_dir_buffer);
+
+    var buffer: [std.fs.max_path_bytes + 1]u8 = undefined;
+    const path = try std.fmt.bufPrintZ(&buffer, "{s}/accounts.db", .{tmp_path});
+
+    var rooted: Rooted = try .init(path);
+    errdefer rooted.deinit();
+
+    const db: @This() = try .init(allocator, rooted);
+    return .{ db, tmp };
+}
+
+pub fn deinit(self: *Db) void {
+    const allocator = self.allocator;
+    self.rooted.deinit();
+    self.unrooted.deinit(allocator);
+}
+
+/// Clones the account shared data.
+pub fn put(self: *Db, slot: Slot, address: Pubkey, data: AccountSharedData) !void {
+    // used for tests, tracking what the largest slot is and
+    // making sure we don't write to something before that.
+    if (builtin.is_test)
+        if (self.rooted.largest_rooted_slot) |lrs|
+            if (lrs >= slot) return error.CannotWriteRootedSlot;
+
+    const cloned = try data.clone(self.allocator);
+    errdefer cloned.deinit(self.allocator);
+    try self.unrooted.put(self.allocator, slot, address, cloned);
+}
+
+// TODO: this should be trivial to put on another thread; this currently blocks the main replay
+// thread for 30-40ms per slot on testnet.
+pub fn onSlotRooted(self: *Db, newly_rooted_slot: Slot) error{FailedToRoot}!void {
+    const rooted_index: *Unrooted.SlotIndex = for (self.unrooted.slots) |*index| {
```

Comment:
```suggestion
    const unrooted_index: *Unrooted.SlotIndex = for (self.unrooted.slots) |*index| {
```

Reply:
This is the SlotIndex for the newly rooted slot. So maybe newly_rooted_index? `unrooted` feels a bit confusing